godtools/Managers/TractManager.swift

```diff
@@ -16,57 +16,64 @@ class TractManager: NSObject {
-        let manifest = xmlData["manifest"]
-        
-        let primaryColorString: String = (manifest.element?.attribute(by: "primary-color")?.text)!
-        let primaryTextColorString: String = (manifest.element?.attribute(by: "primary-text-color")?.text)!
-        let textColorString: String = (manifest.element?.attribute(by: "text-color")?.text)!
-        
+    func loadResource(resource: DownloadedResource, language: Language) -> (pages: [XMLIndexer], colors: TractColors) {
         var pages = [XMLIndexer]()
-        for child in manifest.children {
-            if child.element?.name == "page" {
-                let page = loadPage(child)
-                pages.append(page)
+        let tractColors = TractColors()
+        
+        let translation = resource.getTranslationForLanguage(language)
+        if translation != nil {
+            let manifestPath = translation?.manifestFilename
+            if manifestPath != nil {
+                let xmlData = loadXMLFile(manifestPath!)
+                let manifest = xmlData?["manifest"]
+                
+                if manifest != nil {
+                    let primaryColorString: String = (manifest!.element?.attribute(by: "primary-color")?.text) ?? "rgba(59, 164, 219, 1)"
+                    let primaryTextColorString: String = (manifest!.element?.attribute(by: "primary-text-color")?.text) ?? "rgba(59, 164, 219, 1)"
+                    let textColorString: String = (manifest!.element?.attribute(by: "text-color")?.text) ?? "rgba(90, 90, 90, 1)"
+                    
+                    for child in manifest!["pages"].children {
+                        if child.element?.name == "page" {
+                            let page = loadPage(child)
+                            pages.append(page)
+                        }
+                    }
+                    
+                    tractColors.primaryColor = primaryColorString.getRGBAColor()
+                    tractColors.primaryTextColor = primaryTextColorString.getRGBAColor()
+                    tractColors.textColor = textColorString.getRGBAColor()
+                }
             }
         }
         
-        let tractColors = TractColors()
-        tractColors.primaryColor = primaryColorString.getRGBAColor()
-        tractColors.primaryTextColor = primaryTextColorString.getRGBAColor()
-        tractColors.textColor = textColorString.getRGBAColor()
-        
         return (pages, tractColors)
     }
     
     func loadPage(_ child: XMLIndexer) -> XMLIndexer{
-        let resource = child.element?.attribute(by: "id")?.text
+        let resource = child.element?.attribute(by: "src")?.text
         let page = loadXMLFile(resource!)
-        return page
+        return page!
     }
     
-    func loadXMLFile(_ resource: String) -> XMLIndexer {
+    func loadXMLFile(_ resourcePath: String) -> XMLIndexer? {
+        let documentsPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
```

Comment:
In my latest branch (PR coming) If you extend `GTDataManager` you can access this property on the super class. We can talk about how to do that, possibly in the future.